tests/integration/test_scaling.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import time
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+from scaling_helpers import check_key, get_password, restart_unit, srvr, write_key
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+
+
+async def ping_servers(ops_test: OpsTest):
+    logger.info("pinging servers")
+    for unit in ops_test.model.applications[APP_NAME].units:
+        host = unit.public_address
+        logger.info(f"{srvr(host)=}")
+        mode = srvr(host)["Mode"]
+        assert mode in ["leader", "follower"]
+
+
+@pytest.mark.abort_on_fail
+async def test_deploy_active(ops_test: OpsTest):
+    charm = await ops_test.build_charm(".")
+    await ops_test.model.deploy(charm, application_name=APP_NAME, num_units=3)
+    await ops_test.model.block_until(lambda: len(ops_test.model.applications[APP_NAME].units) == 3)
+    await ops_test.model.set_config({"update-status-hook-interval": "10s"})
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)
+    for unit in range(3):
+        assert ops_test.model.applications[APP_NAME].units[unit].workload_status == "active"
+    await ops_test.model.set_config({"update-status-hook-interval": "60m"})
```

Comment:
There is nothing wrong with this function, but I think this design might be cleaner:
```
@pytest.mark.abort_on_fail
async def test_build_and_deploy(ops_test: OpsTest) -> None:
    """Build and deploy Zookeeper"""
    my_charm = await ops_test.build_charm(".")
    await ops_test.model.deploy(my_charm, num_units=len(UNIT_IDS)) # UNIT_IDS = [0,1,2]
    await ops_test.model.wait_for_idle()


@pytest.mark.abort_on_fail
async def test_status(ops_test: OpsTest) -> None:
    """Verifies that the application and units are active."""
    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)
    assert len(ops_test.model.applications[APP_NAME].units) == len(UNIT_IDS)
```